pkg/cache/placeholder_manager_test.go

```diff
@@ -86,3 +88,67 @@ func TestCreateAppPlaceholders(t *testing.T) {
 	err = placeholderMgr.createAppPlaceholders(app)
 	assert.Error(t, err, "failed to create pod tg-test-group-2-app01-15")
 }
+
+func TestCleanUp(t *testing.T) {
+	const (
+		appID     = "app01"
+		queue     = "root.default"
+		namespace = "test"
+	)
+	mockedContext := initContextForTest()
+	mockedSchedulerAPI := newMockSchedulerAPI()
+	app := NewApplication(appID, queue,
+		"bob", map[string]string{constants.AppTagNamespace: namespace}, mockedSchedulerAPI)
+	mockedContext.applications[appID] = app
+	res := app.getNonTerminatedTaskAlias()
+	assert.Equal(t, len(res), 0)
+
+	pod1 := &v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name: "pod-01",
+			UID:  "UID-01",
+		},
+	}
+	pod2 := &v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name: "pod-02",
+			UID:  "UID-02",
+		},
+	}
+	taskID1 := "task01"
+	task1 := NewTask(taskID1, app, mockedContext, pod1)
+	task1.placeholder = true
+	app.taskMap[taskID1] = task1
+	taskID2 := "task02"
+	task2 := NewTask(taskID2, app, mockedContext, pod2)
+	task2.placeholder = true
```

Comment:
Can we add another task with placeholder flag = false?

Reply:
OK, done.